### PR TITLE
Port heretics from TG

### PR DESCRIPTION
**Pull request by SomeguyManperson**
tgstation/tgstation#51199

tldr:
About The Pull Request
This PR adds a new roundstart antagonist. HERETICS.

Heretics are powerful antagonists whose goal is to reach a final ritual which will cause a horrific effect, based on the path the heretic chooses.

Mechanics
Cultist starts off with 2 items, Codex Cicatrix and Living Heart.

The Codex Cicatrix is similar to a traitor uplink. It allows you to research eldritch knowledge using charges. Eldritch knowledge gives you magic abilities like spells , passive abilities or rituals you can perform on the eldritch rune. You can use other cultist's Codex Cicatrix, but it will show YOUR Knowledge, not his.

The Living Heart is similar to contractor tablet. If you put it on the eldritch rune and activate it will give you a target to kill and sacrifice on the rune, granting you a charge. Charge is given to the first Codex Cicatrix that the function finds on you.

How to get charges
There are 2 ways of acquiring charges for Codex Cicatrix. First one is sacrificing people hunted by the Living Heart.

Second way is through influences.

What is an influence?

As a heretic, if you decide to have a walk you will quickly notice strange objects that have randomized names, like 'rising heat' or 'lowering cold' etc. These are influences, they are only visible to you, and other eldritch cultists. To acquire a charge simple use your book on the small black part of the influence in the middle, and it will harvest it. Be aware that after you harvest the influence it will become visible to everyone, so you have to be sneaky in what you do!

Paths
As a heretic if you open your Codex Cicatrix you will see 5 knowledges, 3 of which you can research. If you look over the 'research' button you will see for example ' Rust path'
or 'Ash path'. These are indicators on which route the knowledge lies. There are 3 main paths:

Ash - Movement, crowd control and being hard to catch.

Rust - Area denial, breaking down anything not living, passive healing

Flesh - Summoning eldritch horrors.

Think of the skill tree like the path of exile one.
It is all inter-connected. You can easily get an ability from rust spend a few points to get one from flesh and then from ash, but remember the cheapest way is to beeline your chosen path, even if it locks you out of many other abilities.

Why It's Good For The Game
new gamemode good

:cl:  Edge, a few Sprites from Kryson, theos for porting
rscadd: Heretics have been spotted near your area, don't let them succeed.
rscadd: Added a sane version of GetAllContents called get_all_gear().
tweak: moved blind to be a pointed spell and adds support for pointed spells
tweak: adds support for cone spells
/:cl:

### CODE REVIEW

**Comment by Tipy1802** [comment]
Epic, it would be cool if the curator got something in his study room to help combat heretics. The study room is supposed to be filled with forbidden/occult knowledge but it’s only used for devil, a game mode that’s not even in rotation, it would give the curator an actual purpose

**Review by Partheo — APPROVED** [review]
Beautiful sprites by Kryson.

**Comment by Jarod1200** [comment]
MEAT SNAKE MEAT SNAKE

**Review by ynot01 — CHANGES_REQUESTED** [review]
there is no antag hud

**Review by Partheo — DISMISSED** [review]
Ping me if there's actually no existing antag hud and I'll make it

**Review by JamieD1 — APPROVED** [review]
Leave me alone adam
